### ISSUE TITLE
Bikeshed seems inconsistent with when method arguments need to be mentioned.

### ISSUE THREAD

**Issue by mkruisselbrink** [tabatkins/bikeshed#634]
In the FileAPI bikeshed conversion there is IDL like this:
```idl
interface Blob {};
interface FileReader {
  void readAsText(Blob blob, optional DOMString label);
};
```

Furthermore there are reference to the method like `{{FileReader/readAsText()}}` and references to the methods arguments like `{{FileReader/readAsText()/blob}}`. The method references link correctly (despite absence of non-optional arguments), but the reference to the arguments complain that no definition is found unless I change them to `{{FileReader/readAsText(blob)/blob}}`. It seems to me like bikeshed should accept both or neither, not the current behavior where sometimes it's okay to leave out the arguments while other times it is not.

**Comment by mkruisselbrink** [comment]
Actually ``{{FileReader/readAsText(blob)/blob}}` worked in my simplified sample, but in the actual spec only ``{{FileReader/readAsText(blob, label)/blob}}` works, presumably because of the explicit argument definition that's also in the spec:
```
<dfn argument for="FileReader/readAsText(blob, label)">blob</dfn>
```

Also unless I'm going crazy, yesterday this same spec compiled without errors, so something seems to have changed recently?

**Comment by tabatkins** [comment]
You are correct, it's inconsistent and not a good situation.  This is because autolinking is ~magic~ and that magic is unreliable in some cases and hard to reason about, even for me.

I'm in the middle of refactoring autolinking right now to make this work a lot better.  I expect progress to be slow and gradual, but to complete the work in the next few weeks.  Apologies, and please bear with me while I make this all act more rational.

**Comment by mkruisselbrink** [comment]
Actually my reduced sample still doesn't seem to match the spec. In my reduced sample (with above IDL and dfn tag) I can link to these method arguments, in the actual spec I don't seem to be able to link to them in any way I can come up with. `{{FileReader/readAsText(blob)/blob}}` fails with `LINK ERROR: No 'idl' refs found for 'blob' with for='FileReader/readAsText(blob)'.`, while `{{FileReader/readAsText(blob, label)/blob}}` fails with `LINK ERROR: No 'idl' refs found for 'readAsText(blob, label)/blob'.` (the "for='FileReader'" seems to have gotten lost somehow?)

**Comment by birtles** [comment]
I'm currently encountering this too with `{{Animatable/animate()/options}}`. That used to work, however I discovered that if I use `{{Animatable/animate(frames, options)/options}}` and debug the autolinking regexp, it is getting tripped up by the spaces here:

  https://github.com/tabatkins/bikeshed/blob/master/bikeshed/shorthands.py#L151

Basically, the first match would consume up to the space in the middle of `frames, options` then backtrack to find the first '/' not the last one. If I drop the spaces before the '}' it works because it correctly ends up with a linkFor of `Animatable/animate(frames, options)` and a linkText of `options`.

I've no idea if there are cases that need that space, however.
